fractal_server/app/routes/api/v2/workflow.py

```diff
@@ -319,8 +319,43 @@ async def import_workflow(
     # Insert tasks
 
     for wf_task in workflow.task_list:
-        source = wf_task.task.source
-        task_id = source_to_id[source]
+        if isinstance(wf_task.task, TaskImportV2Legacy):
+            stm = select(TaskV2).where(TaskV2.source == wf_task.task.source)
+        else:
+            stm = (
+                select(TaskV2)
+                .join(TaskGroupV2)
+                .where(TaskGroupV2.pkg_name == wf_task.task.pkg_name)
+                .where(TaskV2.name == wf_task.task.name)
+            )
+            if wf_task.task.version is not None:
+                stm = stm.where(TaskGroupV2.version == wf_task.task.version)
+            else:
+                stm = stm.order_by(TaskGroupV2.timestamp_created.desc()).limit(
+                    1
+                )
+
+        res = await db.execute(stm)
+        task = res.scalars().all()
+        if task is None:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Expected one TaskV2 to match the imported one.",
+            )
+        elif len(task) == 1:
+            task_id = task[0].id
+
+        elif len(task) != 1:
+            # Option: task that belong to the current user has highest priority
```

Comment:
Disambiguation by ownership:

1. List of tasks is given (query was already executed).
2. If the list includes one tasks that has user_id matching with user.id -> done.
3. Else, if one of the tasks has user_group_id matching with default_group.id (which has to be queried by name - at the beginning of the function) -> done
4. Else, we should take the one associated to the usergroup such that `LinkUserGroup.timestamp_created` is oldest.